waspc/src/Wasp/AppSpec.hs

```diff
@@ -21,6 +29,32 @@ data AppSpec = AppSpec
     externalCodeDirPath :: !(Path' Abs (Dir ExternalCode.SourceExternalCodeDir)),
     -- | Absolute path to the directory in wasp project source that contains database migrations.
     migrationsDir :: Maybe (Path' Abs (Dir DbMigrationsDir)),
+    -- | Absolute path to the .env file in wasp project source. It contains env variables to be
+    -- provided to the server only during the development.
     dotEnvFile :: Maybe (Path' Abs File'),
+    -- | If true, it means project is being compiled for production/deployment -> it is being "built".
+    -- If false, it means project is being compiled for development purposes (e.g. "wasp start").
     isBuild :: Bool
   }
+
+-- TODO: What if this fails (because of `head`)? Returning a Maybe here would be PITA later in the code.
+--   But would be cool if we had an extra step that somehow ensures that app exists and
+--   throws nice error if it doesn't. Some step that validated AppSpec. Maybe we could
+--   have a function that returns `Validated AppSpec` -> so basically smart constructor,
+--   validates AppSpec and returns it wrapped with `Validated`,
+--   and then functions like `getApp` would be executed on `Validated AppSpec`, so we can be sure
+--   they will not return an error (sure in the sense that we know `Validated` checked some stuff already).
+--   For this we should create `newtype ValidAppSpec = ValidAppSpec AppSpec` (or some better name?) and then
+--   function `validateAppSpec :: AppSpec -> ValidAppSpec` and hide constructor `ValidAppSpec`.
+--   But then we will be passing `ValidAppSpec` all around the Generator hm, is that ok?
+--   I think this makes sense, but I should probably postpone it for some future issue, to not overcomplicate things now.
+getApp :: AppSpec -> (String, App)
+getApp spec = head $ takeDecls @App (decls spec)
+
+-- TODO: Make this return "Named" declarations? I already have a comment for this somewhere else I think.
```

Comment:
I should either create a github issue for this, or solve it in this PR.

Still not sure if I should go with `newtype Named a = Named (String, a)` or `type Named a = (String, a)`. First one is typed :D, while the second one is easier to deconstruct.